src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -334,7 +334,325 @@ def assemble_matrix_rhs(
+            self.discretize(g_master, g_slave, data_master, data_slave, data_edge)
+
+        if not g_master.dim == g_slave.dim:
+            raise AssertionError("Slave and master must have same dimension")
+
+        master_ind = 0
+        slave_ind = 1
+
+        # Generate matrix for the coupling. This can probably be generalized
+        # once we have decided on a format for the general variables
+        mg = data_edge["mortar_grid"]
+
+        dof_master = self.discr_master.ndof(g_master)
+        dof_slave = self.discr_slave.ndof(g_slave)
+
+        if not dof_master == matrix[master_ind, master_ind].shape[1]:
+            raise ValueError(
+                """The number of dofs of the master discretization given
+            in FluxPressureContinuity must match the number of dofs given by the matrix
+            """
+            )
+        elif not dof_slave == matrix[master_ind, slave_ind].shape[1]:
+            raise ValueError(
+                """The number of dofs of the slave discretization given
+            in FluxPressureContinuity must match the number of dofs given by the matrix
+            """
+            )
+        elif not self.ndof(mg) == matrix[master_ind, 2].shape[1]:
+            raise ValueError(
+                """The number of dofs of the edge discretization given
+            in FluxPressureContinuity must match the number of dofs given by the matrix
+            """
+            )
+        # We know the number of dofs from the master and slave side from their
+        # discretizations
+        dof = np.array(
+            [
+                matrix[master_ind, master_ind].shape[1],
+                matrix[slave_ind, slave_ind].shape[1],
+                self.ndof(mg),
+            ]
+        )
+        cc = np.array([sps.coo_matrix((i, j)) for i in dof for j in dof])
+        cc_master = cc.reshape((3, 3))
+        cc_slave = cc_master.copy()
+        cc_mortar = cc_master.copy()
+
+        # The rhs is just zeros
+        # EK: For some reason, the following lines were necessary to apease python
+        rhs = np.empty(3, dtype=np.object)
+        rhs[master_ind] = np.zeros(dof_master)
+        rhs[slave_ind] = np.zeros(dof_slave)
+        rhs[2] = data_edge["rhs"]
+
+        # The convention, for now, is to put the master grid information
+        # in the first column and row in matrix, slave grid in the second
+        # and mortar variables in the third
+        # If master and slave is the same grid, they should contribute to the same
+        # row and coloumn. When the assembler assigns matrix[idx] it will only add
+        # the slave information because of duplicate indices (master and slave is the same).
+        # We therefore write the both master and slave info to the slave index.
+        if g_master == g_slave:
+            master_ind = 1
+        else:
+            master_ind = 0
+
+        self.discr_master.assemble_int_bound_displacement_trace(
```

Comment:
Now I see where you get the FVEllipticVectorial stuff from, some of those functions are indeed necessary. Nevertheless, I think those should be moved to mpsa - in particular since it seems unlikely we will invent and implement a new fv discretization for elasticity anytime soon.